src/main/java/frc/robot/subsystems/climber/commands/Pull.java

```diff
@@ -0,0 +1,42 @@
+package frc.robot.subsystems.climber.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+import frc.robot.Constants;
+import frc.robot.subsystems.climber.Climber;
+
+public class Pull extends CommandBase {
+
+    private final Climber climber;
+    private double height;
+
+    public Pull(Climber climber, double height) {
+        this.climber = climber;
+        this.height = height;
+        addRequirements(climber);
+    }
+
+    @Override
+    public void initialize() {
+        if (climber.getStopperMode() == Climber.PistonMode.OPEN)

```

Comment:
This command should only pull the climber. Another command should handle engaging the stopper, and another one will engage the shift to the relevant gear.
I recommend the two commands to be `InstanatCommand`s (as I've seen you've done), and the whole sequence (engage stopper + engage gearbox + climb) to be handled through a command group.